org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -416,86 +451,141 @@ private void terminateProcess(IConsumerProcess<U> process) throws EventException
 	public void run() throws EventException {
 		
 		startJobManager();
+		init();
+		
+		while(isActive()) {
+        	try {
+	            boolean ok = consume();
+	            if (!ok) break;
+	                     		
+        	} catch (Throwable ne) {
+        		boolean stayAlive = processException(ne);
+        		if (stayAlive) {
+        			continue;
+        		} else {
+        			break;
+        		}
+        	}
+
+		}
+	}
+
+	private void init() throws EventException {
+		
+		this.waitTime = 0;
 
 		if (runner!=null) {
 			alive.setAlive(true);
 		} else {
 			throw new EventException("Cannot start a consumer without a runner to run things!");
 		}
 		
-		long waitTime = 0;
-		
 		// We process the paused state
 		PauseBean pbean = getPauseBean(getSubmitQueueName());
 		if (pbean!=null) processPause(pbean); // Might set the pause lock and block on checkPaused().
 		 
-		while(isActive()) {
-        	try {
-        		checkPaused(); // blocks until not paused.
-        		if (!isActive()) break; // Might have pasued for a long time.
-        		
-        		// Consumes messages from the queue.
-	        	Message m = getMessage(uri, getSubmitQueueName());
-	            if (m!=null) {
-		        	waitTime = 0; // We got a message
-	            	
-	            	// TODO FIXME Check if we have the max number of processes
-	            	// exceeded and wait until we don't...
-	            	
-	            	TextMessage t = (TextMessage)m;
-	            	
-	            	final String json  = t.getText();
-	            	
-					@SuppressWarnings("unchecked")
-					final U bean   = (U) service.unmarshal(json, getBeanClass());
-                    
-	            	executeBean(bean);
-	            	
-	            }
-	            
-        	} catch (EventException | InterruptedException ne) {
-        		if (Thread.interrupted()) break;
- 				logger.error("Cannot consume message ", ne);
-       		    if (isDurable()) continue;
-        		break;
-         		
-        	} catch (Throwable ne) {
-        		logger.debug("Error in consumer!", ne);
-        		if (ne.getClass().getSimpleName().contains("Json")) {
-            		logger.error("Fatal except deserializing object!", ne);
-            		continue;
-        		}
-        		if (ne.getClass().getSimpleName().endsWith("UnrecognizedPropertyException")) {
-        			logger.error("Cannot deserialize bean!", ne);
-            		continue;
-        		}
-        		
-        		if (ne.getClass().getSimpleName().endsWith("ClassCastException")) {
-    				logger.error("Problem with serialization?", ne);
-        		}
+		// We should pause if there are things in the queue
+		// This is because on a server restart the user will 
+		// need to choose the visit again and get the baton.
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1181)